modules/stochastic_tools/src/transfers/SerializedSolutionTransfer.C

```diff
@@ -0,0 +1,297 @@
+  const auto & serialized_solution_reporter = getParam<std::string>("solution_container");
+
+  for (MooseIndex(n) i = 0; i < n; i++)
+  {
+
+    if (getFromMultiApp()->hasLocalApp(i))
+    {
+      FEProblemBase & app_problem = getFromMultiApp()->appProblemBase(i);
+
+      std::vector<UserObject *> reporters;
+      app_problem.theWarehouse()
+          .query()
+          .condition<AttribSystem>("UserObject")
+          .condition<AttribName>(serialized_solution_reporter)
+          .queryInto(reporters);
+
+      if (reporters.empty())
+        paramError("solution_container",
+                   "Unable to find reporter with name '",
+                   serialized_solution_reporter,
+                   "'");
+      else if (reporters.size() > 1)
+        paramError("solution_container",
+                   "We found more than one reporter with the name '",
+                   serialized_solution_reporter,
+                   "'");
+
+      _solution_container.push_back(dynamic_cast<SolutionContainer *>(reporters[0]));
+
+      if (!_solution_container[i - _sampler_ptr->getLocalRowBegin()])
+        paramError("solution_container",
+                   "The parallel storage reporter is not of type 'SolutionContainer'");
+    }
+  }
+}
+
+void
+SerializedSolutionTransfer::initializeInBatchMode()
+{
+  // First we fetch the solution containers from the subapps. This function is used
+  // in batch mode only so we will have one solution container on each rank
+  _solution_container.clear();
+
+  const auto & serialized_solution_reporter = getParam<std::string>("solution_container");
+
+  FEProblemBase & app_problem = getFromMultiApp()->appProblemBase(_app_index);
+
+  std::vector<UserObject *> reporters;
+  app_problem.theWarehouse()
+      .query()
+      .condition<AttribSystem>("UserObject")
+      .condition<AttribName>(serialized_solution_reporter)
+      .queryInto(reporters);
+
+  if (reporters.empty())
+    paramError("solution_container",
+               "Unable to find reporter with name '",
+               serialized_solution_reporter,
+               "'");
+  else if (reporters.size() > 1)
+    paramError("solution_container",
+               "We found more than one reporter with the name '",
+               serialized_solution_reporter,
+               "'");
+
+  _solution_container.push_back(dynamic_cast<SolutionContainer *>(reporters[0]));
+
+  if (!_solution_container[0])
+    paramError("solution_container",
+               "The parallel storage reporter is not of type 'SolutionContainer'");
+}
+
+void
+SerializedSolutionTransfer::execute()
+{
+  initializeInNormalMode();
+
+  const dof_id_type n = getFromMultiApp()->numGlobalApps();
+
+  for (MooseIndex(n) i = 0; i < n; i++)
+  {
+    if (getFromMultiApp()->hasLocalApp(i))
+    {
+      FEProblemBase & app_problem = getFromMultiApp()->appProblemBase(i);
+      NonlinearSystemBase & nl = app_problem.getNonlinearSystemBase();
```

Comment:
so this doesnt support auxvariables from subapps?

Reply:
Correct. I might add that capability in another PR.